astropy/stats/funcs.py

```diff
@@ -757,26 +760,31 @@ def median_absolute_deviation(a, axis=None):
     mad_std
     """
 
-    # Check if the array has a mask and if so use np.ma.median
-    # See https://github.com/numpy/numpy/issues/7330 why using np.ma.median
-    # for normal arrays should not be done (summary: np.ma.median always
-    # returns an masked array even if the result should be scalar). (#4658)
-    if isinstance(a, np.ma.MaskedArray):
-        func = np.ma.median
-    else:
-        func = np.median
+    if func is None:
+        # Check if the array has a mask and if so use np.ma.median
+        # See https://github.com/numpy/numpy/issues/7330 why using np.ma.median
+        # for normal arrays should not be done (summary: np.ma.median always
+        # returns an masked array even if the result should be scalar). (#4658)
+        if isinstance(a, np.ma.MaskedArray):
+            func = np.ma.median
+        else:
+            func = np.median
 
     a = np.asanyarray(a)
     a_median = func(a, axis=axis)
 
     # broadcast the median array before subtraction
     if axis is not None:
-        a_median = np.expand_dims(a_median, axis=axis)
+        if isinstance(axis, (tuple, list)):
```

Comment:
Ah fun, lists are valid for `np.median` (docstring speaks of a *sequence of int*) but not for `np.mean`. I usually prefer to use `try/except` for real errors, here using `isinstance` is more explicit imo.
And thanks for restarting 😉